packages/components/molecules/f-rating/src/components/_tests/Rating.test.js

```diff
@@ -1,10 +1,117 @@
-import { shallowMount } from '@vue/test-utils';
+import { createLocalVue, shallowMount } from '@vue/test-utils';
+import { VueI18n } from '@justeat/f-globalisation';
 import VRating from '../Rating.vue';
+import i18n from './helpers/setup';
+
+const localVue = createLocalVue();
+localVue.use(VueI18n);
 
 describe('Rating', () => {
+    let propsData;
+    let wrapper;
+    const $tc = jest.fn();
+
+    beforeEach(() => {
+        propsData = {
+            starRating: 2,
+            maxStarRating: 5
+        };
+        wrapper = shallowMount(VRating, {
+            propsData,
+            localVue,
+            i18n,
+            mocks: {
+                $tc
+            }
+        });
+    });
+
     it('should be defined', () => {
-        const propsData = {};
-        const wrapper = shallowMount(VRating, { propsData });
         expect(wrapper.exists()).toBe(true);
     });
+
+    describe('methods', () => {
+        describe('`hasRating`', () => {
+            it('should exist', () => {
+                expect(wrapper.vm.hasRating).toBeDefined();
+            });
+
+            it('should contain a description `c-rating-description`', () => {
+                // Act
+                const result = wrapper.find('[data-test-id="c-rating-description"]');
+
+                // Assert
+                expect(result).toMatchSnapshot();
+            });
+
+            describe('when invoked', () => {
+                it('should return truthy when `starRating` is less than or equal to `rating`', () => {
+                    // Act
+                    const result = wrapper.vm.hasRating(2);
```

Comment:
NIT: It's a little hard to know what's going on in the test because we're 'Arranging' a lot of stuff outside of each test (props etc) - it would be duplication but perhaps easier to understand each test if we created a new wrapper for every test.

Reply:
I think it's ok at the moment :) What is difficult to understand? I'm happy to explain on a call but essentially we only have one set up at the top which will require the translations mock, a vue instance (because we need to tell vue about the translations) & props. I do agree it makes it easier to move it into each describe but only when it makes sense, I think it's quite straightforward here, if we ended up modifying the props data or wrapper in each test but having the setup right at the top then  it makes it harder to follow and therefore moving into each test unit would be better.